src/Symfony/Bundle/FrameworkBundle/FrameworkBundle.php

```diff
@@ -97,6 +98,7 @@ public function build(ContainerBuilder $container)
         $container->addCompilerPass(new LoggingTranslatorPass());
         $container->addCompilerPass(new AddCacheWarmerPass());
         $container->addCompilerPass(new AddCacheClearerPass());
+        $container->addCompilerPass(new CachePoolPrunerPass());
```

Comment:
this pass should be run just before CachePoolClearerPass, so that Definition are resolved and you can check the class of each pool to check if it implements the new interface, and also to allow the removal of unused private cache pool (right now, the registration for the pruner prevents that).